pkg/deploy/envParams.go

```diff
@@ -8,6 +8,59 @@ import (
 	"strings"
 )
 
+func parseParams(fileName string) (map[string]string, error) {
+	paramsEnv, err := os.Open(fileName)
+	if err != nil {
+		return nil, err
+	}
+	defer paramsEnv.Close()
+
+	paramsEnvMap := make(map[string]string)
+	scanner := bufio.NewScanner(paramsEnv)
+	for scanner.Scan() {
+		line := scanner.Text()
+		parts := strings.SplitN(line, "=", 2)
+		if len(parts) == 2 {
+			paramsEnvMap[parts[0]] = parts[1]
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+
+	return paramsEnvMap, nil
+}
+
+func writeParamsToTmp(params map[string]string, tmpDir string) (string, error) {
+	tmp, err := os.CreateTemp(tmpDir, "params.env-")
+	if err != nil {
+		return "", err
+	}
+	defer tmp.Close()
+
+	// Write the new map to temporary file
+	writer := bufio.NewWriter(tmp)
+	for key, value := range params {
+		if _, err := fmt.Fprintf(writer, "%s=%s\n", key, value); err != nil {
+			return "", err
+		}
+	}
+	if err := writer.Flush(); err != nil {
+		fmt.Printf("Failed to write to file: %v", err)
+		return "", err
+	}
+
+	return tmp.Name(), nil
+}
+
+func updateMap(m *map[string]string, key, val, format string) {
+	old := (*m)[key]
+	if old != val {
+		fmt.Printf(format, key, old, val)
```

Comment:
this is going to print something like:
`/opt/manifests/components/path/params.env:'odh-kuberay-operator-controller-image': replacing 'quay.io/repo/image:stable' with 'newregistry/repo/someimage/sometag'`
is it too much? 
my point is, it is useful for "devel mode", but do we really want to print this out for "production"?

and i hope the reason to make a new function updateMap is not for the fmt.Printf() part, 
e.g just  `(*m)[key] = val `isnt it enough even in the case old value and new value are the same? ( the previous code paramsEnvMap[i] = relatedImageValue )

Reply:
> this is going to print something like: `/opt/manifests/components/path/params.env:'odh-kuberay-operator-controller-image': replacing 'quay.io/repo/image:stable' with 'newregistry/repo/someimage/sometag'` is it too much? my point is, it is useful for "devel mode", but do we really want to print this out for "production"?
> 
> and i hope the reason to make a new function updateMap is not for the fmt.Printf() part,

It is :) Otherwise, as you mentioned below, it's not needed.

Well, I was thinking from the point of view of the issue, where we suspect that params.env is not updated.
It could be possible to use logger with levels, but it's anyway does not help to debug production issue if the message is for debug level only.

Should I remove the output or change it?

>  e.g just `(*m)[key] = val `isnt it enough even in the case old value and new value are the same? ( the previous code paramsEnvMap[i] = relatedImageValue )

yes

Reply:
such a tricky question :D 
personally, i prefer to not print too much.
on the other side, i dont know how to track when "bug" happens.

maybe we skip print for now, and try to see if any other way to get to know what is the problem. 
on the worst case, can add this later. wdyt?

Reply:
Fine for me. This patch should fix the issue with race on updating, if there is any.